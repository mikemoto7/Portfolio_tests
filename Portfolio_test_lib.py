
from test_flags import _product_called_from_test
_product_called_from_test = True


import sys
sys.path.append("../lib_for_test")

from general_test_lib import do_test_runstring
import general_test_lib

general_test_lib.product_dir_path = "../Portfolio"







