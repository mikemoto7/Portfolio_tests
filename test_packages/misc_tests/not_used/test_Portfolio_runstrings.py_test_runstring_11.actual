
Processing: user_runstring = ./Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml

Processing: --price_web_update

Processing:  --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing: --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing:  --pt lp_diff

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                    Before                         After                        Before                        After                        Before                        After
Asset                Tot_%    Asset                Tot_%    Asset                Ret_%    Asset               Ret_%    Asset                Non_%    Asset               Non_%
Small_Value           2.24    Small_Value           2.24    Mid_Blend_Aggress     0.00    Mid_Blend_Aggress    0.00    Global                2.73    Global               2.73
Small_Blend           2.66    Small_Blend           2.66    Small_Value           0.55    Small_Value          0.55    Large_Value           4.31    Large_Value          4.31
Healthcare            3.04    Healthcare            3.04    Healthcare            0.81    Healthcare           0.81    Small_Growth          4.50    Small_Growth         4.50
Large_Blend           5.13    Large_Blend           5.13    Small_Blend           0.85    Small_Blend          0.85    Small_Value           4.80    Small_Value          4.80
Large_Value           5.56    Large_Value           5.56    Large_Blend           4.53    Large_Blend          4.53    Mid_Growth            4.97    Mid_Growth           4.97
Mid_Value             6.28    Mid_Value             6.28    Mid_Value             5.89    Mid_Value            5.89    Small_Blend           5.42    Small_Blend          5.42
Global                6.79    Global                6.79    Large_Value           6.38    Large_Value          6.38    Energy                5.46    Energy               5.46
Mid_Blend_Aggress     8.55    Mid_Blend_Aggress     8.55    Global                9.47    Global               9.47    Large_Blend           6.02    Large_Blend          6.02
Small_Growth          9.67    Small_Growth          9.67    Small_Growth         13.08    Small_Growth        13.08    Healthcare            6.43    Healthcare           6.43
Mid_Growth           10.36    Mid_Growth           10.36    Mid_Growth           13.91    Mid_Growth          13.91    Mid_Value             6.87    Mid_Value            6.87
Energy               10.68    Energy               10.68    Energy               14.11    Energy              14.11    Large_Growth          9.21    Large_Growth         9.21
REIT                 12.57    REIT                 12.57    REIT                 14.56    REIT                14.56    REIT                  9.53    REIT                 9.53
Large_Growth         13.21    Large_Growth         13.21    Large_Growth         15.85    Large_Growth        15.85    Mid_Blend_Aggress    21.52    Mid_Blend_Aggress   21.52
Individual_Stocks     3.27    Individual_Stocks     3.27    Individual_Stocks     0.00    Individual_Stocks    0.00    Individual_Stocks     8.22    Individual_Stocks    8.22
Totals              100.01                        100.01                         99.99                        99.99                         99.99                        99.99

