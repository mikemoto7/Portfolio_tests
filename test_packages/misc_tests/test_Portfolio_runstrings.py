import pytest

from Portfolio_test_lib import do_test_runstring


# @pytest.mark.usefixtures("oneTimeSetUp", "setUp")
@pytest.mark.usefixtures()
class TestPortfolio():

    # @pytest.fixture(autouse=True)
    # def classSetup(self, setup):
    #     pass



    # Testcases interactive:

    '''
    def test_runstring_cl(self):
        do_test_runstring("Portfolio.py --cl")

    def test_runstring_cl_l(self):
        do_test_runstring("Portfolio.py --cl l")

    def test_runstring_cl_7(self):
        do_test_runstring("Portfolio.py --cl 7")

    def test_runstring_cl_debug_1(self):
        do_test_runstring("Portfolio.py --cl 7 --debug 1")
    '''


    # Testcases from docstring:


    def test_runstring_price_web_update(self):
        do_test_runstring("Portfolio.py --price_web_update --pfile demo_data_file.xml")

    def test_runstring_ac_web_update(self):
        do_test_runstring("Portfolio.py --ac_web_update --pfile demo_data_file.xml")

    def test_runstring_stats(self):
        do_test_runstring("Portfolio.py --stats --pfile demo_data_file.xml")

    def test_runstring_xml(self):
        do_test_runstring("Portfolio.py --xml --pfile demo_data_file.xml")

    def test_runstring_price_web_update_xml(self):
        do_test_runstring("Portfolio.py --price_web_update --xml --pfile demo_data_file.xml")

    def test_runstring_pt_show(self):
        do_test_runstring("Portfolio.py --pt show --pfile demo_data_file.xml")

    def test_runstring_pt_show_percent(self):
        do_test_runstring("Portfolio.py --pt show_percent --pfile demo_data_file.xml")

    def test_runstring_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --pt show_ac_misc_totals --pfile demo_data_file.xml")

    def test_runstring_pt_lp(self):
        do_test_runstring("Portfolio.py --pt lp --pfile demo_data_file.xml")

    def test_runstring_pt_add(self):
        do_test_runstring("Portfolio.py --pt add:'Small Value',Non_Retirement,50000:'Small Value',Retirement,20000:Healthcare,Retirement,50000 --pfile demo_data_file.xml")

    def test_runstring_inv_show(self):
        do_test_runstring("Portfolio.py --inv show --pfile demo_data_file.xml")

    def test_runstring_inv_diff_1file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml")

    def test_runstring_inv_diff_2file(self):
        do_test_runstring("Portfolio.py --inv diff --pfile demo_data_file.xml1,demo_data_file.xml2")

    def test_runstring_inv_lots(self):
        do_test_runstring("Portfolio.py --inv lots --pfile demo_data_file.xml")

    def test_runstring_all_not_silent(self):
        do_test_runstring("Portfolio.py --all --pfile demo_data_file.xml")

    def test_runstring_all_silent(self):
        do_test_runstring("Portfolio.py --all --silent --pfile demo_data_file.xml")

    def test_runstring_noindent(self):
        do_test_runstring("Portfolio.py --noindent  --all  --pfile demo_data_file.xml")

    def test_runstring_h(self):
        do_test_runstring("Portfolio.py --h")

    def test_runstring_xmlfile(self):
        do_test_runstring("Portfolio.py --xmlfile  --pfile demo_data_file.xml")

    def test_runstring_pdt(self):
        do_test_runstring("Portfolio.py --pdt --pfile demo_data_file.xml")

    def test_runstring_debug_1(self):
        do_test_runstring("Portfolio.py --debug 1  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_2(self):
        do_test_runstring("Portfolio.py --debug 2  --all  --pfile demo_data_file.xml")

    def test_runstring_debug_12(self):
        do_test_runstring("Portfolio.py --debug 12  --all  --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_show(self):
        do_test_runstring("Portfolio.py --price_web_update --inv show --pfile demo_data_file.xml")

    def test_runstring_price_web_update_ac_web_update_pt_show(self):
        do_test_runstring("Portfolio.py --price_web_update --ac_web_update --pt show%lp%add:Small_Value,Retirement,13000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_lp(self):
        do_test_runstring("Portfolio.py --price_web_update --pt lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Large_Blend,Retirement,13000%show_percent%lp --pfile demo_data_file.xml")

    def test_runstring_price_web_update_inv_diff(self):
        do_test_runstring("Portfolio.py --price_web_update --inv diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_percent_bigger_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_percent%lp%add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%show_percent%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_add(self):
        do_test_runstring("Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml")

    def test_runstring_price_web_update_pt_show_ac_misc_totals(self):
        do_test_runstring("Portfolio.py --price_web_update --pt show_ac_misc_totals --pfile demo_data_file.xml")






