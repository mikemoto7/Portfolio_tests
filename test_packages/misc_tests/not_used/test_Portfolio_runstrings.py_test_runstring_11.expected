
Processing: user_runstring = ./Portfolio.py --price_web_update --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000%lp_diff --pfile demo_data_file.xml

Processing: --price_web_update

Processing:  --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing: --pt add:Small_Growth,Non_Retirement,30000:Energy,Non_Retirement,15000:Large_Value,Non_Retirement,15000:Global,Non_Retirement,15000:REIT,Non_Retirement,15000:Healthcare,Non_Retirement,15000

Processing:  --pt lp_diff

What to invest in next = has the lowest percents = at the top of the list--
----------------------------------------------------------------------------
List includes Bonds and Individual_Stocks but they are pushed down to the bottom of the lists.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                    Before                        After                        Before                        After                        Before                         After
Asset                Tot_%    Asset               Tot_%    Asset                Ret_%    Asset               Ret_%    Asset                Non_%    Asset                Non_%
Small_Value           2.24    Small_Value          2.24    Mid_Blend_Aggress     0.00    Mid_Blend_Aggress    0.00    Global                2.73    Global                2.73
Small_Blend           2.66    Small_Blend          2.66    Small_Value           0.55    Small_Value          0.55    Large_Value           4.30    Large_Value           4.30
Healthcare            3.04    Healthcare           3.04    Healthcare            0.81    Healthcare           0.81    Small_Growth          4.49    Small_Growth          4.49
Large_Blend           5.13    Large_Blend          5.13    Small_Blend           0.84    Small_Blend          0.84    Small_Value           4.80    Small_Value           4.80
Large_Value           5.55    Large_Value          5.55    Large_Blend           4.55    Large_Blend          4.55    Mid_Growth            4.95    Mid_Growth            4.95
Mid_Value             6.28    Mid_Value            6.28    Mid_Value             5.88    Mid_Value            5.88    Small_Blend           5.41    Small_Blend           5.41
Global                6.76    Global               6.76    Large_Value           6.38    Large_Value          6.38    Energy                5.46    Energy                5.46
Mid_Blend_Aggress     8.56    Mid_Blend_Aggress    8.56    Global                9.43    Global               9.43    Large_Blend           6.02    Large_Blend           6.02
Small_Growth          9.67    Small_Growth         9.67    Small_Growth         13.09    Small_Growth        13.09    Healthcare            6.42    Healthcare            6.42
Mid_Growth           10.33    Mid_Growth          10.33    Mid_Growth           13.88    Mid_Growth          13.88    Mid_Value             6.88    Mid_Value             6.88
Energy               10.69    Energy              10.69    Energy               14.14    Energy              14.14    Large_Growth          9.17    Large_Growth          9.17
REIT                 12.66    REIT                12.66    REIT                 14.68    REIT                14.68    REIT                  9.59    REIT                  9.59
Large_Growth         13.14    Large_Growth        13.14    Large_Growth         15.76    Large_Growth        15.76    Mid_Blend_Aggress    21.53    Mid_Blend_Aggress    21.53
Individual_Stocks     3.28    Individual_Stocks    3.28    Individual_Stocks     0.00    Individual_Stocks    0.00    Individual_Stocks     8.25    Individual_Stocks     8.25
Totals               99.99                        99.99                         99.99                        99.99                        100.00                        100.00

